src/main/java/org/dependencytrack/policy/cel/CelPolicyEngine.java

```diff
@@ -451,6 +454,10 @@ private static org.dependencytrack.proto.policy.v1.Component mapToProto(final Co
                         .setBlake2B384(trimToEmpty(projection.blake2b_384))
                         .setBlake2B512(trimToEmpty(projection.blake2b_512))
                         .setBlake3(trimToEmpty(projection.blake3));
+        if (projection.getPublishedAt() != null) {
+            componentBuilder.setCurrentVersionLastModified(Timestamp.newBuilder().setSeconds(projection
```

Comment:
You can use [`Timestamps#fromDate`](https://javadoc.io/doc/com.google.protobuf/protobuf-java-util/3.20.1/com/google/protobuf/util/Timestamps.html#fromDate-java.util.Date-) here instead. Besides better readability, it also retains the precision correctly.